### PR TITLE
Add cabal-doctest and idris Setup.hs as smoke tests for BC

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [x] The documentation has been updated, if necessary.
* [x] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by 23Skidoo** [comment]
Good idea!

**Comment by phadej** [comment]
Ah, idris script breaks on windows too.

Note: I'd  add Agda's Setup.hs too, seems that its `Setup.hs` is under MIT license. Is my interpretation of https://hackage.haskell.org/package/Agda-2.5.3/src/LICENSE correct, @hvr 

What other non-trivial custom setups we should be aware off?

I answer above:  `ghc-paths` - I'll add that one a bit later today.

File: Cabal/Cabal.cabal [hunk 1]
```diff
@@ -483,6 +483,20 @@ test-suite check-tests
   ghc-options: -Wall
   default-language: Haskell2010
 
+test-suite custom-setup-tests
```
Comment (23Skidoo) [inline Cabal/Cabal.cabal]: You also need `default-language: Haskell2010` here, that's why CI is red.

**Comment by 23Skidoo** [comment]
Feel free to merge once it's green.

File: Cabal/Cabal.cabal (hunk 1 above)
Comment (23Skidoo) [reply Cabal/Cabal.cabal]: Oh kewl, looks like I can push to this branch myself.

**Comment by 23Skidoo** [comment]
> What other non-trivial custom setups we should be aware off?

Maybe something from Kmett-verse?

**Comment by phadej** [comment]
cabal-doctest is used in a far share of Kmett-verse. I'll check if `lens` copies doc image manually still

**Comment by 23Skidoo** [comment]
There's a failure on 7.4.2, because `IdrisSetup` doesn't support versions that old.

**Comment by 23Skidoo** [comment]
Also `gen-extra-source-files` apparently doesn't take `hs-source-dirs` into account.

**Comment by 23Skidoo** [comment]
See also https://github.com/23Skidoo/all-custom-setups.

**Comment by 23Skidoo** [comment]
@phadej I pushed some patches, it's green now (except one macOS timeout).

**Comment by phadej** [comment]
Great! let's merge and add more later.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Pull request merged**